### ISSUE TITLE
atom lint errors when not using standard diagnostics

### ISSUE THREAD

**Issue by bradennapier** [prettier/prettier-atom#298]
This doesn't appear to actually stop things from working, but it does produce a lot of errors in console on every save.

When using something other than default linter (like is the case if you use the new atom-ide features), every save will start causing an error:

```
~/.atom/packages/prettier-atom/dist/linterInterface/index.js:25 prettier-atom attempted to set messages with linter package, but was unable. Messages: []
```

**Comment by natterstefan** [comment]
@bradennapier I have the same issue, did you solve it?

**Comment by robwise** [comment]
should be fixed in #321

**Comment by burdakovd** [comment]
I'm seeing the same when editing remote file with Nuclide (every time on save I get this error, and can't achieve format-on-save functionality).

I've applied https://github.com/prettier/prettier-atom/pull/321 locally, and the error still persists.

After adding some more logs around the line where the error is being logged, I see that it was caused by file path being undefined rather than by Linter being null.

The error is logged here: https://github.com/prettier/prettier-atom/blob/e8e8e3fc4da30082a8d70f341b729c46a7771823/src/linterInterface/index.js#L19

In my scenario messages appear to be `[]`, filePath is `undefined`, and delegate is an instance of `IndieLinterDelegate`.

Further, editor from which undefined file path is extracted is a TextEditor that contains remote file in there. Is remote editing something that is supported at all for Prettier?
![image](https://user-images.githubusercontent.com/5193161/33798643-4fd0b96e-dd14-11e7-832e-baccd2cf5a7a.png)
